roles/variables/vars/main.yml

```diff
@@ -209,7 +375,7 @@ kafka_broker_properties:
   sasl_gssapi:
     enabled: "{{ 'GSSAPI' in kafka_broker_sasl_enabled_mechanisms }}"
     properties:
-      sasl.kerberos.service.name: "{{kerberos_kafka_broker_primary}}"
+      sasl.kerberos.service.name: "{{kerberos_kafka_controller_primary if kraft_enabled|bool else kerberos_kafka_broker_primary}}"
```

Comment:
Why are we adding `kafka_controller` cluster name in broker's service name?

Reply:
because kerberos sasl service is kerberos_kafka_controller_primary and for it to talk to broker, broker should have the same service name in properties

Reply:
Let's discuss more on this.
I believe if broker want to connect to controller via Kerberos auth, it would need those properties under `listener.name.controller.sasl.kerberos.service.name`